### PR TITLE
Remove ArrayViews dependency

### PR DESCRIPTION
**Pull request by ararslan**
This uses the Base functionality for array views: `sub` on 0.4 and `view` on 0.5. This change might hurt performance due to the concerns mentioned in the ArrayViews README regarding `SubArray` versus `ArrayView` as well as the fact that `Base.reshape` makes a copy and `ArrayViews.reshape_view` does not. AFAIK there's no Base analog for `reshape_view`.

This PR is kind of an experiment to see what happens and provides an alternative to #178 for fixing the current 0.5 errors caused by the `Base.view`/`ArrayViews.view` conflict.

cc @nalimilan @simonbyrne

### CODE REVIEW

**Comment by ararslan** [comment]
Awesome, both this and #178 work! Now it's just a matter of determining which is the better choice, or whether there's an even better alternative.

**Comment by ararslan** [comment]
Bump. We should get the 0.5 tests passing, especially as 0.5 is nearing a release candidate. At present, our choices are this or #182. Seems like @simonbyrne, @andreasnoack, and myself are in favor of removing the dependency on ArrayViews. @nalimilan? Anyone else want to weigh in here?

File: src/StatsBase.jl [hunk 1]
```diff
@@ -2,8 +2,8 @@ __precompile__(true)
 
 module StatsBase
     using Compat
-    import Compat.String
-    using ArrayViews
+    import Compat: String, view
```
Comment (nalimilan) [inline src/StatsBase.jl]: I think `using` is enough here, as you don't overload these functions (I know `import` was here before, and I've been guilty of this too).

**Comment by nalimilan** [comment]
I have no opinion on the matter, so if other says it's OK I'm all for it. Just squash and merge now that "you've got the power". BTW, feel free to tag a new release too.

**Comment by ararslan** [comment]
Speaking of which, thanks so much for the invitation to the organization, @nalimilan! I'm honored! :heart:

**Comment by nalimilan** [comment]
I must say that wasn't my own decision, it was a consensus among JuliaStats members. Welcome on board!

**Comment by ararslan** [comment]
Ah, well thank you all JuliaStats members! 😄

File: src/StatsBase.jl (hunk 1 above)
Comment (ararslan) [inline src/StatsBase.jl]: Looks like the explicit `import` actually is necessary here; it fails on 0.4 because it can't find `view`.

**Comment by ararslan** [comment]
Gotta love `git reset --hard`. The CI didn't even need to run again. 😄

File: src/StatsBase.jl (hunk 1 above)
Comment (nalimilan) [inline src/StatsBase.jl]: Maybe I wasn't explicit enough. My proposal was `using Compat: String, view`. Or maybe I'm just wrong.

Comment (ararslan) [inline src/StatsBase.jl]: Oh, I didn't even realize that syntax existed. Well, I just merged...

**Pull request merged**